src/cryptography/hazmat/primitives/ciphers/modes.py

```diff
@@ -51,7 +51,7 @@ def nonce(self) -> bytes:
 
 class ModeWithAuthenticationTag(metaclass=abc.ABCMeta):
     @abc.abstractproperty
-    def tag(self) -> bytes:
+    def tag(self) -> typing.Optional[bytes]:
```

Comment:
I would like to explicitly point at this change for inspection.

This change is necessary due to the `GCM` class, where the tag is in fact optional. If the parent class defines a non-optional bytes, mypy will rightfully complain.

I'm not an expert enough on the details of the underlying algorithm if the hint added here is accurate enough. We could also add an override instead or make bytes non-optional in that one subclass.

Reply:
I think Optional[bytes] is probably as good as we can do here without redesigning the (not very good) API.